File: 21. Phyton/21. Modificadores de acceso/Modificadores de acceso.py

class Clase:

    # Definición de atributo publico (Se peude acceder fuera de la clase)
    atributo1 = valor1

    # definición de atributo protegido (Solo se puede acceder desde la clase y las clases heredadas)
    _atributo2 = valor2

    # definición de atributo privado (Solo se puede acceder desde la clase)
    __atributo3 = valor3

    # Definición de metodo publico (Se peude acceder fuera de la clase)
    def metodo1(self, atributo1, atributo2, atributoN):
        # Contenido

    ## definición de metodo protegido (Solo se puede acceder desde la clase y las clases heredadas)
    def _metodo2(self, atributo1, atributo2, atributoN):
        # Contenido

    # Definición de metodo privado (Solo se puede acceder desde la clase)
    def __metodo3(self, atributo1, atributo2, atributoN):
        # Contenido

File: 01. Lenguajes de proposito general/04. Paradigmas/02. Orientada a objetos/02. Modificadores de acceso/Java/Modificadores de acceso.java
package paquete1;

// Clase principal (Debe tener el mismo nombre del archivo)
public class Clase {
   
    // Alcance publico
    // - Desde la misma clase:                  Si
    // - Desde otra clase dento del package:    Si
    // - Desde otra clase fuera del package:    Si
    // - Subclase en dento del propio package:  Si
    // - Subclase en fuera del propio package:  Si
    public tipo variable;       // Atributo
    public tipo metodo(){};     // Método
    
    // Alcance protejido
    // - Desde la misma clase:                  Si
    // - Desde otra clase dento del package:    Si
    // - Desde otra clase fuera del package:    No
    // - Subclase en dento del propio package:  Si
    // - Subclase en fuera del propio package:  Si
    protected tipo variable;       // Atributo
    protected tipo metodo(){};     // Método
    
    // Alcance por default
    // - Desde la misma clase:                  Si
    // - Desde otra clase dento del package:    Si
    // - Desde otra clase fuera del package:    No
    // - Subclase en dento del propio package:  Si
    // - Subclase en fuera del propio package:  No
    tipo variable;       // Atributo
    tipo metodo(){};     // Método
    
    // Alcance privado
    // - Desde la misma clase:                  Si
    // - Desde otra clase dento del package:    No
    // - Desde otra clase fuera del package:    No
    // - Subclase en dento del propio package:  No
    // - Subclase en fuera del propio package:  No
    private tipo variable;       // Atributo
    private tipo metodo(){};     // Método
    
}

// Clase segundaria (Puede tener cualquier nombre) (Puede exisitir un archivo de solo clases segundarias)
class Clase {
    // Contenido
}